asserts/asserts.go

```diff
@@ -528,8 +551,15 @@ func assembleAndSign(assertType *AssertionType, headers map[string]interface{},
 	finalHeaders["type"] = assertType.Name
 	finalHeaders["body-length"] = strconv.Itoa(bodyLength)
 
-	if _, err := checkNotEmptyString(finalHeaders, "authority-id"); err != nil {
-		return nil, err
+	if withAuthority {
+		if _, err := checkNotEmptyString(finalHeaders, "authority-id"); err != nil {
+			return nil, err
+		}
+	} else {
+		_, ok := finalHeaders["authority-id"]
+		if ok {
+			return nil, fmt.Errorf("freestanding %q assertion cannot have authority-id set", assertType.Name)
```

Comment:
freestanding doesn't really mean anything even to us, and it definitely won't be understandable by the user.